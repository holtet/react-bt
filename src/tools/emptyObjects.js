export const malt = {
  comments: '',
  grams: 0,
  boughtDate: new Date(),
  maltDefinition: { name: '', description: '', manufacturer: '', country: '' },
};
export const maltType = {
  name: '',
  country: '',
  manufacturer: '',
  description: '',
};
export const yeast = {
  grams: 0,
  packages: 0,
  comments: '',
  boughtDate: new Date(),
  yeastType: { name: '', format: '' },
};
export const equipment = {
  boughtDate: new Date(),
  comments: '',
  amount: 0,
  equipmentType: {
    name: '',
    description: '',
  },
};
export const hop = {
  grams: 0,
  year: 2017,
  alpha: 0.0,
  comments: '',
  boughtDate: new Date(),
  hopType: { name: '', format: '' },
};
export const hopType = {
  name: '',
  manufacturer: '',
  country: '',
  description: '',
};
