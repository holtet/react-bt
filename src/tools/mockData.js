const yeastTypes = [
  {
    name: 'Saf5',
    manufacturer: 'Safale',
    country: 'England',
    format: 'DRY',
    _links: {
      self: {
        href: 'http://localhost:8080/yeastTypes/14',
      },
      yeastType: {
        href: 'http://localhost:8080/yeastTypes/14',
      },
    },
  },
  {
    name: 'WLP400',
    manufacturer: 'WyEast',
    country: 'Belgium',
    format: 'LIQUID',
    _links: {
      self: {
        href: 'http://localhost:8080/yeastTypes/15',
      },
      yeastType: {
        href: 'http://localhost:8080/yeastTypes/15',
      },
    },
  },
];
const hopTypes = [
  {
    name: 'Cascade',
    manufacturer: 'US Hop Co',
    country: 'USA',
    format: 'PELLETS',
    _links: {
      self: {
        href: 'http://localhost:8080/hopTypes/10',
      },
      hopType: {
        href: 'http://localhost:8080/hopTypes/10',
      },
    },
  },
  {
    name: 'hallertau',
    manufacturer: 'Hops GMBH',
    country: 'Germany',
    format: 'PELLETS',
    _links: {
      self: {
        href: 'http://localhost:8080/hopTypes/11',
      },
      hopType: {
        href: 'http://localhost:8080/hopTypes/11',
      },
    },
  },
];
const hops = [
  {
    year: 2017,
    alpha: 3.7,
    grams: 90,
    boughtDate: '2020-05-29',
    comments: '',
    openedDate: null,
    hopType: {
      name: 'Cascade',
      manufacturer: 'US Hop Co',
      country: 'USA',
      format: 'PELLETS',
    },
    type: {
      name: 'Cascade',
      manufacturer: 'US Hop Co',
      country: 'USA',
      format: 'PELLETS',
    },
    _links: {
      self: {
        href: 'http://localhost:8080/hops/12',
      },
      hop: {
        href: 'http://localhost:8080/hops/12{?projection}',
        templated: true,
      },
      type: {
        href: 'http://localhost:8080/hops/12/type',
      },
    },
  },
  {
    year: 2018,
    alpha: 5.3,
    grams: 15,
    boughtDate: '2020-05-29',
    comments: '',
    openedDate: null,
    hopType: {
      name: 'hallertau',
      manufacturer: 'Hops GMBH',
      country: 'Germany',
      format: 'PELLETS',
    },
    _links: {
      self: {
        href: 'http://localhost:8080/hops/13',
      },
      hop: {
        href: 'http://localhost:8080/hops/13{?projection}',
        templated: true,
      },
      type: {
        href: 'http://localhost:8080/hops/13/type',
      },
    },
  },
];
const apiHops = { _embedded: { hops: hops } };
const newHop = { id: null, comments: '' };
module.exports = {
  hopTypes,
  hops,
  apiHops,
  newHop,
  yeastTypes,
};
