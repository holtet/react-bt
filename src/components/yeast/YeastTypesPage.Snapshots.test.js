import React from 'react';
import YeastTypesPageView from './YeastTypesPageView';
import renderer from 'react-test-renderer';
import { yeastTypes } from '../../tools/mockData';
import { BrowserRouter as Router } from 'react-router-dom';

it('has correct snapshot', () => {
  const tree = renderer.create(
    <Router>
      <YeastTypesPageView
        types={yeastTypes}
        onNewItemClick={jest.fn()}
        onDeleteClick={jest.fn()}
        redirectToTypePage={false}
      />
    </Router>
  );
  expect(tree).toMatchSnapshot();
});

it('has correct snapshot with redirect', () => {
  const tree = renderer.create(
    <Router>
      <YeastTypesPageView
        types={yeastTypes}
        onNewItemClick={jest.fn()}
        onDeleteClick={jest.fn()}
        redirectToTypePage={true}
      />
    </Router>
  );
  expect(tree).toMatchSnapshot();
});
