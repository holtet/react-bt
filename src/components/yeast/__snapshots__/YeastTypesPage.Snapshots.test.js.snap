// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has correct snapshot 1`] = `
Array [
  <table
    className="table table-sm"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Manufacturer
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <a
            href="/yeasttype/0"
            onClick={[Function]}
          >
            Saf5
          </a>
        </td>
        <td>
          Safale
        </td>
        <td>
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/yeasttype/1"
            onClick={[Function]}
          >
            WLP400
          </a>
        </td>
        <td>
          WyEast
        </td>
        <td>
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>,
  <button
    className="btn btn-primary"
    onClick={[MockFunction]}
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    Add
  </button>,
]
`;

exports[`has correct snapshot with redirect 1`] = `
Array [
  <table
    className="table table-sm"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Manufacturer
        </th>
        <th />
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <a
            href="/yeasttype/0"
            onClick={[Function]}
          >
            Saf5
          </a>
        </td>
        <td>
          Safale
        </td>
        <td>
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <a
            href="/yeasttype/1"
            onClick={[Function]}
          >
            WLP400
          </a>
        </td>
        <td>
          WyEast
        </td>
        <td>
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            Delete
          </button>
        </td>
      </tr>
    </tbody>
  </table>,
  <button
    className="btn btn-primary"
    onClick={[MockFunction]}
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    Add
  </button>,
]
`;
